bokeh/embed/bundle.py

```diff
@@ -59,66 +76,72 @@
 # Dev API
 #-----------------------------------------------------------------------------
 
+class Artifact:
+    pass
 
-class ScriptRef:
-    def __init__(self, url, type="text/javascript"):
+class ScriptRef(Artifact):
+    def __init__(self, url: str, type: str = "text/javascript"):
         self.url = url
         self.type = type
 
 
-class Script:
-    def __init__(self, content, type="text/javascript"):
+class Script(Artifact):
+    def __init__(self, content: str, type: str = "text/javascript"):
         self.content = content
         self.type = type
 
 
-class StyleRef:
-    def __init__(self, url):
+class StyleRef(Artifact):
+    def __init__(self, url: str):
         self.url = url
 
 
-class Style:
-    def __init__(self, content):
+class Style(Artifact):
+    def __init__(self, content: str):
         self.content = content
 
 
 class Bundle:
-    @classmethod
-    def of(cls, js_files, js_raw, css_files, css_raw, hashes):
```

Comment:
@mattpap Any reason this was removed here without a note? I do find this method a bit weird (and superfluous) but since it was part of the API and used in a few places Panel used it and it's now breaking. Happy to see this removed as a proper deprecation but not as part of a PR which is meant to be purely about typing.

Reply:
Actually don't worry about it, I'll just patch it in Panel. A heads up when you change APIs (even of the internals) would still be appreciated.

Reply:
It's such an obscure API that I didn't bother to check if panel doesn't use it (I usually do check).

Reply:
On a side note, I restored `dict` support in `Subcommand`'s argument handling (in https://github.com/bokeh/bokeh/pull/11290/commits/3f9e82d83e9a3590bedcb477830f74fb1a4c5488).

Reply:
Definitely obscure but got copied over when I adapted some of the bundling code. Easy fix though.

Reply:
And would never have caused real issues in a proper release since Panel is now strict in its pinning (i.e. it in pins `<2.4`)